contracts/volt/VoltV2.sol

```diff
@@ -0,0 +1,440 @@
+        return
+            nCheckpoints > 0 ? checkpoints[account][nCheckpoints - 1].votes : 0;
+    }
+
+    /// @notice Determine the prior number of votes for an account as of a block number
+    /// @dev Block number must be a finalized block or else this function will revert to prevent misinformation.
+    /// @param account The address of the account to check
+    /// @param blockNumber The block number to get the vote balance at
+    /// @return The number of votes the account had as of the given block
+    function getPriorVotes(address account, uint256 blockNumber)
+        public
+        view
+        returns (uint256)
+    {
+        require(blockNumber < block.number, "Volt: not yet determined");
+
+        uint32 nCheckpoints = numCheckpoints[account];
+        if (nCheckpoints == 0) {
+            return 0;
+        }
+
+        // First check most recent balance
+        if (checkpoints[account][nCheckpoints - 1].fromBlock <= blockNumber) {
+            return checkpoints[account][nCheckpoints - 1].votes;
+        }
+
+        // Next check implicit zero balance
+        if (checkpoints[account][0].fromBlock > blockNumber) {
+            return 0;
+        }
+
+        uint32 lower = 0;
+        uint32 upper = nCheckpoints - 1;
+        while (upper > lower) {
+            uint32 center = upper - (upper - lower) / 2; // ceil, avoiding overflow
+            Checkpoint memory cp = checkpoints[account][center];
+            if (cp.fromBlock == blockNumber) {
+                return cp.votes;
+            } else if (cp.fromBlock < blockNumber) {
+                lower = center;
+            } else {
+                upper = center - 1;
+            }
+        }
+        return checkpoints[account][lower].votes;
+    }
+
+    function _burn(address src, uint256 amount) internal {
+        require(balances[src] >= amount, "Volt: burn amount exceeds balance");
+
+        totalSupply -= amount;
+        balances[src] -= amount;
+
+        emit Transfer(src, address(0), amount);
+
+        _moveDelegates(delegates[src], address(0), amount);
+    }
+
+    function _spendAllowance(address src, uint256 amount) internal {
+        address spender = msg.sender;
+        uint256 spenderAllowance = allowances[src][spender];
+
+        if (spender != src && spenderAllowance != type(uint256).max) {
+            uint256 newAllowance = spenderAllowance - amount;
+            allowances[src][spender] = newAllowance;
+
+            emit Approval(src, spender, newAllowance);
+        }
+    }
+
+    function _delegate(address delegator, address delegatee) internal {
+        address currentDelegate = delegates[delegator];
+        uint256 delegatorBalance = balances[delegator];
+        delegates[delegator] = delegatee;
+
+        emit DelegateChanged(delegator, currentDelegate, delegatee);
+
+        _moveDelegates(currentDelegate, delegatee, delegatorBalance);
+    }
+
+    function _transferTokens(
```

Comment:
add natspec comment to this function, explaining who calls, and the function caller's assumption.